base/show.jl

```diff
@@ -336,8 +336,10 @@ Show an expression and result, returning the result.
 macro show(exs...)
     blk = Expr(:block)
     for ex in exs
+        sym = gensym()
+        push!(blk.args, :($(sym) = begin value=$(esc(ex)) end))
```

Comment:
This could just use the existing `value` variable.

Reply:
I didn't understand why there was `begin value=$(esc(ex)) end` instead of just `$(esc(ex))` in the first place.

Do you mean it should generate code like
```
value = x
show(value)
value = y
show(value)
```
?

Reply:
Bump @JeffBezanson when you get a chance to take a look.

Reply:
Yes, that would be ok, or
```
sym1 = x
show(sym1)
sym2 = y
show(sym2)
sym2
```

Reply:
That was just how the old version captured `value` in order to return it. There's no reason to do two assignments per expression.